src/importer/DND5SpellImporter.ts

```diff
@@ -34,34 +34,68 @@ export class DND5SpellImporter {
 
     public importSpellsByUrl = async (url: string): Promise<void> => {
         try {
-            const response: multiSpellImport = await axios.get(url);
-            this.importSpellsByData(response.results);
+            axios.get(url).then(response => {
+                const spellLocations: multiSpellImport = response.data;
+                const spells: spellImport[] = [];
+                // collect errors that occur during the import
+                const errorList: string[] =[];
+                spellLocations.results.forEach(async location => {
+                    try {
+                        axios.get(`http://dnd5eapi.co${location.url}`).then(resp => {
+                        spells.push(resp.data);
+                        // check if last spell received
+                        if ((spellLocations.results.length + errorList.length) == spells.length) {
+                            if (errorList.length) {
+                                console.warn('Errors during spell import: ',errorList);
+                            }
+                            this.importSpellsByData(spells);
+                        }
+                    });
+                    } catch (error) {
+                        errorList.push(`An error occured while getting ${location.url}.`);
+                    }
+                });
+            });
         } catch (error) {
             console.error('While importing spells form url', url, 'an error occured.\n', error);
         }
     };
 
     public importSpellsByData = (spells: spellImport[]): void => {
-        const dnd5Spells = spells.map(spell => {
-            return this.mapToDND5Spell(spell)
-        });
+        const dnd5Spells = spells.map(this.mapToDND5Spell);
         DND5Spell.bulkCreate(dnd5Spells, { updateOnDuplicate: this.spellKeys });
     };
 
-    private mapToDND5Spell = (spell: spellImport): object => {
-        return {
+    private mapToDND5Spell = (spell: spellImport): spellData => {
+
+        const spellData: spellData = {
             name: spell.name,
-            description: spell.desc?.join(),
-            higherLevelsDescription: spell.higher_level?.join(),
+            description: '',
+            higherLevelsDescription: '',
             range: spell.range,
-            components: spell.components?.join(),
+            components: '',
             ritual: spell.ritual,
             duration: spell.duration,
             concentration: spell.concentration,
             castingTime: spell.casting_time,
-            school: spell.school?.name,
-            materials: spell.material,
+            school: '',
+            materials: spell.material? spell.material: '',
             level: spell.level
+        };
+
+        if (spell.desc) {
+            spellData.description = spell.desc.join();
+        }
+        if (spell.higher_level) {
+            spellData.higherLevelsDescription = spell.higher_level?.join();
```

Comment:
higher level war nicht nullable??? Oke das war meine eigene Doofheit das ding kann nullable sein in der DB.

Reply:
Ich muss das trotzdem so holen, weil ich nicht join auf null aufrufen kann;)

Reply:
Joa ich hab auch eher gemeint das ich dumm war und das DB feld auch null sein darf eigentlich